### ISSUE TITLE
Better error messages when setting piece size of transmission-create

### ISSUE THREAD

**Issue by ieee8023** [transmission/transmission#866]
The help says this:

```
 -s --piecesize <size in KiB> Set how many KiB each piece should be, overriding
                              the preferred default
```

So I run this with 16000KB thinking it should be fine

```
$ transmission-create -s 16000 test 
WARNING: no trackers specified
Creating torrent "/home/debian/test.torrent" ...[2019-03-29 14:20:29.735] Failed to set piece size to 15.62 MiB, leaving it at 32.00 KiB
```

but the piece size needed to be a power of 2:

```
$ transmission-create -s 16384 test 
WARNING: no trackers specified
Creating torrent "/home/debian/test.torrent" .... done!
```

I think the error and help text should say "piece size must be a power of 2".

**Comment by carandraug** [comment]
I don't think that is the issue. Attempting to set piecesize above 2MiB, even if power of two, gives a similar error:

    transmission-create   --piecesize 4194304   PATH-FILE 
    Creating torrent 
    ...
    Failed to set piece size to 0.00 KiB, leaving it at 2.00 MiB

I think that transmission-create just has an inherent limit on piece size of 2**21 bytes.

**Comment by i30817** [comment]
This is kind of dumb not being mentioned in the man page, but you need to 'simply' using the appropriate suffix with the number argument of -s. 

if you want megabytes uses -s 16MiB and the error mysteriously disappears.
